graphene_sqlalchemy/filters.py

```diff
@@ -44,10 +44,29 @@ def execute_filters(cls: Type[FieldFilter], query, filter_dict: Dict) -> Tuple[Q
         clauses = []
         for field, filt_dict in filter_dict.items():
             model = cls._meta.model
-            field_filter_type: FieldFilter = cls._meta.fields[field]._type
-            model_field = getattr(model, field)
-            query, _clauses = field_filter_type.execute_filters(query, model_field, filt_dict)
-            clauses.extend(_clauses)
+            # Relationships are Dynamic, we need to resolve them fist
+            # Maybe we can cache these dynamics to improve efficiency
+            # Check with a profiler is required to determine necessity
+            input_field = cls._meta.fields[field]
+            if isinstance(input_field, graphene.Dynamic):
+                field_filter_type = input_field.get_type().type
+            else:
+                field_filter_type = cls._meta.fields[field].type
+            # TODO we need to save the relationship props in the meta fields array
```

Comment:
We need to make sure to explicitly alias every joined model 
https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#orm-query-join-aliased-true-from-joinpoint-removed